xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -6154,6 +6180,33 @@ send_attach_req(xlator_t *this, struct rpc_clnt *rpc, char *path,
     brick_req.dict.dict_val = NULL;
     brick_req.dict.dict_len = 0;
 
+    /* Set graceful_cleanup key in dicitonary to check the key value at the time
+       of handling a detach rpc request at brick side
+    */
+    if (graceful_cleanup) {
+        dict = dict_new();
+        if (!dict) {
+            gf_smsg(this->name, GF_LOG_ERROR, errno, GD_MSG_DICT_CREATE_FAIL,
+                    NULL);
+            ret = -ENOMEM;
+            goto *errlbl;
```

Comment:
Not sure how much effort this requires, but I would try to avoid indirect `goto` jumps. Makes the code much harder to follow. If it's not trivial, we can leave this for a future patch, though.

Reply:
done